locals.tf

```diff
@@ -1,4 +1,45 @@
 locals {
+  # Manage certificates
+  pre_install_certificate_1 = (
+    length(var.runners_gitlab_certificate) > 0
+    ? <<-EOT
+      mkdir -p /etc/gitlab-runner/certs/
+      cat <<- EOF > /etc/gitlab-runner/certs/gitlab.crt
+      ${var.runners_gitlab_certificate}
+      EOF
+    EOT
+    : ""
+  )
+  pre_install_certificate_2 = (
+    length(var.runners_ca_certificate) > 0
+    ? <<-EOT
+      mkdir -p /etc/gitlab-runner/certs/
+      cat <<- EOF > /etc/gitlab-runner/certs/ca.crt
+      ${var.runners_ca_certificate}
+      EOF
+    EOT
+    : ""
+  )
+  pre_install_certificates_all = join("\n", [
+    local.pre_install_certificate_1,
+    local.pre_install_certificate_2
+    ]
+  )
+  pre_install_certificates_end = <<-EOT
+    chmod 600 /etc/gitlab-runner/certs/*.crt
+    chmod -R a+r /etc/gitlab-runner
+    cp /etc/gitlab-runner/certs/*.crt /etc/pki/ca-trust/source/anchors
+    update-ca-trust extract
+  EOT
+  pre_install_certificates = (
+    length(local.pre_install_certificates_all) > 5 # Newlines
```

Comment:
suggestion: we could quickly run into problems here with the hardcoded `5`. What about something like `length(var.runners_gitlab_certificate) +length(var.runners_ca_certificate) > 0`? Then you could also get rid of the `pre_install_certificates_all` variable.